tests/integration/cocoon/test_cocoon_instruments.py

```diff
@@ -742,3 +742,110 @@ def test_load_instrument_lookthrough(self, _, df, mapping):
+        :param str file_name: The name of the test data file
+        :param dict{str, str} mapping_required: The dictionary mapping the dataframe fields to LUSID's required base transaction/holding schema
+        :param dict{str, str} mapping_optional: The dictionary mapping the dataframe fields to LUSID's optional base transaction/holding schema
+        :param dict{str, str} identifier_mapping: The dictionary mapping of LUSID instrument identifiers to identifiers in the dataframe
+        :param list[str] property_columns: The columns to create properties for
+        :param str properties_scope: The scope to add the properties to
+        :param any expected_outcome: The expected outcome
+
+        :return: None
+        """
+
+        data_frame = pd.read_csv(Path(__file__).parent.joinpath(file_name))
+
+        responses = cocoon.cocoon.load_from_data_frame(
+            api_factory=self.api_factory,
+            scope=scope,
+            data_frame=data_frame,
+            mapping_required=mapping_required,
+            mapping_optional=mapping_optional,
+            file_type="instruments",
+            identifier_mapping=identifier_mapping,
+            property_columns=property_columns,
+            properties_scope=properties_scope,
+        )
+
+        self.assertEqual(
+            0,
+            len(responses["instruments"]["errors"]),
+            responses["instruments"]["errors"],
+        )
+
+        failed = [
+            f
+            for response in responses["instruments"]["success"]
+            for f in response.failed.values()
+        ]
+
+        self.assertEqual(0, len(failed), failed)
+
+        self.assertEqual(
+            first=sum(
+                [
+                    len(response.values)
+                    for response in responses["instruments"]["success"]
+                ]
+            ),
+            second=len(data_frame["client_internal"].unique()),
+        )
+
+        # Assert that by no unmatched_identifiers are returned in the response for instruments
+        self.assertFalse(responses["instruments"].get("unmatched_identifiers", False))
+
+        combined_successes = {
+            correlation_id: instrument
+            for response in responses["instruments"]["success"]
+            for correlation_id, instrument in response.values.items()
+        }
+
+        self.assertTrue(
+            expr=all(
+                sorted(combined_successes[key].properties, key=lambda p: p.key)
+                == sorted(value.properties, key=lambda p: p.key)
+                for key, value in expected_outcome["instruments"].values.items()
+            )
+        )
+
+        # Assert that instruments have been assigne scope
+        self.assertTrue(
+            expr=all(
+                instrument.scope == scope
+                for response in responses["instruments"]["success"]
+                for instrument in response.values.values()
+            )
+        )
```

Comment:
Sonar is not enjoying this test because most of it is a duplicate of Lines220-272. 

I think you might be able to just add in L844+ to `test_load_from_data_frame_instruments_success` and you will have what you need.